crates/exex/exex/src/context.rs

```diff
@@ -92,4 +93,29 @@ impl<Node: FullNodeComponents> ExExContext<Node> {
     pub fn task_executor(&self) -> &TaskExecutor {
         self.components.task_executor()
     }
+
+    /// Replaces notifications stream with a stream of notifications without a head.
+    ///
+    /// Consumes the type and returns a new instance of [`ExExContext`].
+    pub fn with_notifications_without_head(self) -> Self {
```

Comment:
isn't this the default?

Can we come up with a less confusing name? :)

Reply:
yep, removed these methods on the `ExExContext` and just left the mutating versions